packages/babel-plugin-proposal-pipeline-operator/src/fsharpVisitor.js

```diff
@@ -0,0 +1,58 @@
+import { types as t } from "@babel/core";
+
+const fsharpVisitor = {
+  BinaryExpression(path) {
+    const { scope } = path;
+    const { node } = path;
+    const { operator, left } = node;
+    let right = node.right.body;
+    if (operator !== "|>") return;
+
+    let optimizeArrow =
+      t.isArrowFunctionExpression(right) &&
+      t.isExpression(right.body) &&
+      !right.async &&
+      !right.generator;
+    let param;
+
+    if (optimizeArrow) {
+      const { params } = right;
+      if (params.length === 1 && t.isIdentifier(params[0])) {
+        param = params[0];
+      } else if (params.length > 0) {
+        optimizeArrow = false;
+      }
+    } else if (t.isIdentifier(right, { name: "eval" })) {
+      right = t.sequenceExpression([t.numericLiteral(0), right]);
+    }
+
+    if (optimizeArrow && !param) {
+      // Arrow function with 0 arguments
+      path.replaceWith(t.sequenceExpression([left, right.body]));
+      return;
+    }
```

Comment:
This `optimizeArrow` logic is duplicated between minimal & fsharp. Can we dedupe?

Reply:
Will take a look at that.